mobile-save-for-later/src/main/scala/com/gu/sfl/identity/IdentityService.scala

```diff
@@ -71,4 +86,25 @@ class IdentityServiceImpl(identityConfig: IdentityConfig, okHttpClient: OkHttpCl
     })
     promise.future
   }
+
+  def userFromRequestOauth(identityHeaders: IdentityHeader, requiredScope: List[IdentityAccessScope]): Future[Option[String]] = {
+    val promise = Promise[Option[String]]
+    val claims = oktaLocalValidator.claimsFromAccessToken(identityHeaders.auth.stripPrefix("Bearer "), requiredScope)
+    println(claims.toString)
+    claims match {
+      case Left(e) => {
+        promise.failure(OktaValidationException(e))
+      }
+      case Right(c) => promise.success(Option(c.identityId))
+    }
+
+    promise.future
+  }
```

Comment:
It doesn't look like this method needs to return a Future.  It could just be an unwrapped call to `oktaLocalValidator`.
```
def userFromRequestOauth(identityHeaders: IdentityHeader, requiredScope: List[IdentityAccessScope]): Option[String] =
  oktaLocalValidator.claimsFromAccessToken(identityHeaders.auth.stripPrefix("Bearer "), requiredScope).toOption
```